packages/in_app_purchase_tizen/lib/src/in_app_purchase_tizen_platform.dart

```diff
@@ -0,0 +1,165 @@
+  ///
+  /// This field should not be used out of test code.
+  @visibleForTesting
+  late final BillingManager billingManager;
+
+  static final Set<String> _productIdsToConsume = <String>{};
+
+  @override
+  Future<bool> isAvailable() async {
+    return billingManager.isAvailable();
+  }
+
+  @override
+  Future<ProductDetailsResponse> queryProductDetails(
+      Set<String> identifiers) async {
+    ProductsListApiResult response;
+    PlatformException? exception;
+    _identifiers = identifiers;
+    _appId = identifiers.toList()[0];
+    _serverType = identifiers.toList()[5];
+    try {
+      response = await billingManager.requestProducts(identifiers.toList());
+    } on PlatformException catch (e) {
+      exception = e;
+      response = const ProductsListApiResult(
+          cPStatus: 'fail to receive response',
+          cPResult: 'fail to receive response',
+          checkValue: 'fail to receive response',
+          totalCount: 0,
+          itemDetails: <ItemDetails>[]);
+    }
+
+    List<SamsungCheckoutProductDetails> productDetailsList =
+        <SamsungCheckoutProductDetails>[];
+
+    productDetailsList = response.itemDetails
+        .map((ItemDetails productWrapper) =>
+            SamsungCheckoutProductDetails.fromProduct(productWrapper))
+        .toList();
+    final List<String> invalidMessage = <String>[];
+    switch (response.cPStatus) {
+      case '100000':
+        break;
+      default:
+        invalidMessage.add(response.toJson().toString());
+    }
+
+    final ProductDetailsResponse productDetailsResponse =
+        ProductDetailsResponse(
+      productDetails: productDetailsList,
+      notFoundIDs: invalidMessage,
+      error: exception == null
+          ? null
+          : IAPError(
+              source: kIAPSource,
+              code: exception.code,
+              message: exception.message ?? '',
+              details: exception.details),
+    );
+    return productDetailsResponse;
+  }
+
+  @override
+  Future<void> restorePurchases({
+    String? applicationUserName,
+  }) async {
+    List<GetUserPurchaseListAPIResult> responses;
+
+    responses = await Future.wait(<Future<GetUserPurchaseListAPIResult>>[
+      billingManager.requestPurchases(
+          applicationUserName, _identifiers.toList())
+    ]);
+
+    final List<PurchaseDetails> pastPurchases =
+        responses.expand((GetUserPurchaseListAPIResult response) {
+      return response.invoiceDetails;
+    }).map((InvoiceDetails purchaseWrapper) {
+      final SamsungCheckoutPurchaseDetails purchaseDetails =
+          SamsungCheckoutPurchaseDetails.fromPurchase(purchaseWrapper);
+
+      purchaseDetails.status = PurchaseStatus.restored;
+      return purchaseDetails;
+    }).toList();
+    _purchaseUpdatedController.add(pastPurchases);
+  }
+
+  @override
+  Future<bool> buyNonConsumable({required PurchaseParam purchaseParam}) async {
+    final BillingBuyData billingResultWrapper = await billingManager.buyItem(
```

Comment:
According to the documentation, this call should add a new event to `purchaseStream` when the purchase is complete. As far as I can see, `purchaseStream` is currently only updated on `restorePurchases`.

Reply:
Fixed.
It will add the new purchase into `purchaseStream` when the `payResult` of `buyItem` is 'SUCCESS'.
[PurchaseUpdate if buyitem successfully](https://github.com/flutter-tizen/plugins/pull/562/commits/48ed9da0788db764fd3cb1bf16ade54732bb161d)